mapie/estimator/estimator.py

```diff
@@ -0,0 +1,505 @@
+            self.k_ = np.full(
+                shape=(n_samples, cv.get_n_splits(X, y)),
+                fill_value=np.nan,
+                dtype=float,
+            )
+            if self.method == "naive":
+                estimators_ = [single_estimator_]
+            else:
+                estimators_ = Parallel(self.n_jobs, verbose=self.verbose)(
+                    delayed(self._fit_oof_estimator)(
+                        clone(estimator), X, y, train_index, sample_weight
+                    )
+                    for train_index, _ in cv.split(X)
+                )
+            if isinstance(cv, ShuffleSplit):
+                single_estimator_ = estimators_[0]
+
+        self.single_estimator_ = single_estimator_
+        self.estimators_ = estimators_
+
+        return self
+
+    def predict(
+        self,
+        X: ArrayLike,
+        ensemble: bool = False,
+        return_multi_pred: bool = True
+    ) -> Union[NDArray, Tuple[NDArray, NDArray, NDArray]]:
+        """
+        Predict target from X. It also computes the prediction per train sample
+        for each test sample according to ``self.method``.
+
+        Parameters
+        ----------
+        X: ArrayLike of shape (n_samples, n_features)
+            Test data.
+
+        ensemble: bool
+            Boolean determining whether the predictions are ensembled or not.
+            If ``False``, predictions are those of the model trained on the
+            whole training set.
+            If ``True``, predictions from perturbed models are aggregated by
+            the aggregation function specified in the ``agg_function``
+            attribute.
+
+            If ``cv`` is ``"prefit"`` or ``"split"``, ``ensemble`` is ignored.
+
+            By default ``False``.
+
+        return_multi_pred: bool
+
+
+        Returns
+        -------
+        Tuple[NDArray, NDArray, NDArray]
+            - Predictions
+            - The multiple predictions for the lower bound of the intervals.
+            - The multiple predictions for the upper bound of the intervals.
+        """
+        check_is_fitted(self, self.fit_attributes)
+
+        y_pred = self.single_estimator_.predict(X)
+
+        if self.method in self.no_agg_methods_ or self.cv in self.no_agg_cv_:
+            y_pred_multi_low = y_pred[:, np.newaxis]
+            y_pred_multi_up = y_pred[:, np.newaxis]
+        else:
+            y_pred_multi = self._pred_multi(X)
+
+            if self.method == "minmax":
+                y_pred_multi_low = np.min(y_pred_multi, axis=1, keepdims=True)
+                y_pred_multi_up = np.max(y_pred_multi, axis=1, keepdims=True)
+            else:
+                y_pred_multi_low = y_pred_multi
+                y_pred_multi_up = y_pred_multi
+
+            if ensemble:
+                y_pred = aggregate_all(self.agg_function, y_pred_multi)
+
+        if return_multi_pred:
+            return y_pred, y_pred_multi_low, y_pred_multi_up
+        else:
+            return y_pred
```

Comment:
UPD: it is possible to compure `y_pred_multi_low, y_pred_multi_up` but not return them. This is a suggestion to manage this problem:
```suggestion
        y_pred = self.single_estimator_.predict(X)
        if not return_multi_pred and not ensemble:
            return y_pred
        
        if self.method in self.no_agg_methods_ or self.cv in self.no_agg_cv_:
            y_pred_multi_low = y_pred[:, np.newaxis]
            y_pred_multi_up = y_pred[:, np.newaxis]
        else:
            y_pred_multi = self._pred_multi(X)

            if self.method == "minmax":
                y_pred_multi_low = np.min(y_pred_multi, axis=1, keepdims=True)
                y_pred_multi_up = np.max(y_pred_multi, axis=1, keepdims=True)
            else:
                y_pred_multi_low = y_pred_multi
                y_pred_multi_up = y_pred_multi

            if ensemble:
                y_pred = aggregate_all(self.agg_function, y_pred_multi)

        if return_multi_pred:
            return y_pred, y_pred_multi_low, y_pred_multi_up
        else:
            return y_pred
```

Reply:
I don't understand what "this problem" is.
In the method `get_bounds` of `ConformityScore` we need `y_pred_multi_up` and `y_pred_multi_low` to compute the bounds. So we need to return them.